azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java

```diff
@@ -294,7 +296,19 @@ private User getCognitoUser(final String token) throws UserManagerException {
     }
     Map<String, Claim> claims = decodedToken.getClaims();
     User user = null;
-    user = new User(claims.get("cognito:username").asString());
+    String username = null;
+    if (claims.containsKey("preferred_username")) {
+      username = claims.get("preferred_username").asString();
```

Comment:
You still need to suffix the first 3 chars of sub

Reply:
Should be sorted now